lib/heuristics/concerns/scheduling_data_initialisation.rb

```diff
@@ -21,26 +21,21 @@
 module SchedulingDataInitialization
   extend ActiveSupport::Concern
 
+  def reduce_to_24h(time, day_index)
+    time < 84600 ? time : time - day_index * 86400
+  end
+
   def generate_route_structure(vrp)
     vrp.vehicles.each{ |vehicle|
       capacity = compute_capacities(vehicle[:capacities], true)
-      vrp.units.reject{ |unit| capacity.has_key?(unit[:id]) }.each{ |unit| capacity[unit[:id]] = 0.0 }
+      vrp.units.reject{ |unit| capacity.key?(unit[:id]) }.each{ |unit| capacity[unit[:id]] = 0.0 }
       @candidate_routes[vehicle.original_id][vehicle.global_day_index] = {
-        vehicle_id: vehicle.id,
-        global_day_index: vehicle.global_day_index,
-        tw_start: (vehicle.timewindow.start < 84600) ? vehicle.timewindow.start : vehicle.timewindow.start - vehicle.global_day_index * 86400,
-        tw_end: (vehicle.timewindow.end < 84600) ? vehicle.timewindow.end : vehicle.timewindow.end - vehicle.global_day_index * 86400,
-        start_point_id: vehicle.start_point&.id,
-        end_point_id: vehicle.end_point&.id,
+        vehicle: vehicle,
+        tw_start: reduce_to_24h(vehicle.timewindow.start, vehicle.global_day_index),
+        tw_end: reduce_to_24h(vehicle.timewindow.end, vehicle.global_day_index),
```

Comment:
why not just do `vehicle.timewindow.start % 84600` we wouldn't even need a function that way.